### PR TITLE
Guava Table support and smaller fixes

### PR DESCRIPTION
**Pull request by maraswrona**
**TL;DR;**

I noticed that the Guava's [Table](https://github.com/google/guava/wiki/NewCollectionTypesExplained#table) structure is not directly supported by this library. I need this in my project, so I decided to share my work.

**What's inside:**
- serializers for all guava's `Table` implementations, including: `ArrayTable`, `HashBasedTable`, `TreeBasedTable` and `ImmutableTable`
- a fix to `TreeMultimapSerializer` to include key and value comparators (this allows to serialize a `TreeMultimap` together with non-standard comparators)
- a fix to `ImmutableMapSerializer` to include some private implementations used by `DenseImmutableTable`

**Details**

So I'm using kryo-serializers + subzero + hazelcast + guava. I noticed I couldn't serialize the `ImmutableTable` properly so I decided to investigate why. Surprisingly some of the Table implementations are serialized just fine (e.g. `HashBasedTable`) but that is only because they are handled by the generic `FieldSerializer`, so their private fields are basically inferred and serialized. The reason why this doesn't work for `ImmutableTable` implementation is that it internally uses some private implementation of `ImmutableMap` (e.g. `Column` or `Row`) which in turn are serialized by the `FieldSerializer` as regular `Map`, and fail during deserialization. 

This PR fixes both sides of this problem: it adds direct support for `Table` implementations (including `ImmutableTable`) as well as registers additional private implementations of `ImmutableMap` with `ImmutableMapSerializer`.

Additionally I noticed that the `TreeMultimapSerializer` didn't cover non-default comparators. I added support for those in `TreeBasedTableSerializer`, so decided to fix it for `TreeMultimapSerializer` too. 

Please take a look and review before merging. I'll appreciate any feedback!

### CODE REVIEW

**Comment by magro** [comment]
Awesome! Can you please also update the readme?

**Comment by maraswrona** [comment]
I updated the readme. 

I also changed the implementation of Table serialization strategy. The first attempt was to serialize a collection of cells as R + C + V triplets. I've run some benchmarks and the output produced by this approach is 10% bigger than what you get from the default FieldSerializer. It's especially inefficient for very dense tables which are filled e.g. 100%. Then basically we serialize rows and columns multiple times unnecessarily.

Now the strategy goes through the rows and through the columns for each row in nested loop and saves row first, and then its colums and values. It is 5% more efficient than the FieldSerializer and 12% more efficient than the previous implementation.

I keep quite a lot of these Table objects in my Hazelcast cluster, so it matters to me if they don't occupy to much memory, so I wanted to measure this. I hope I didn't make any mistakes.
